src/com/google/enterprise/gsafeed/GsaFeedFileSender.java

```diff
@@ -109,12 +111,23 @@ private void buildPostParameter(StringBuilder sb, String name,
     return toEncodedBytes("" + sb);
   }
 
+  /**
+   * Builds the multipart HTTP message for uploading group definitions.
+   * @throws NullPointerException if feedtype is null.
+   */
   private byte[] buildGroupsXmlMessage(String groupsource, String feedtype,
       String xmlDocument) {
     StringBuilder sb = new StringBuilder();
-    buildPostParameter(sb, "groupsource", "text/plain", groupsource);
-    buildPostParameter(sb, "feedtype", "text/plain", feedtype);
-    buildPostParameter(sb, "data", "text/xml", xmlDocument);
+    String ft = feedtype.toLowerCase(US);
+    if (ft.equals("incremental")) {
+      buildPostParameter(sb, "groupsource", "text/plain", groupsource);
+      buildPostParameter(sb, "feedtype", "text/plain", feedtype);
+      buildPostParameter(sb, "data", "text/xml", xmlDocument);
+    } else if (ft.equals("cleanup")) {
```

Comment:
Why just "cleanup" and not "full" and "replace" as well?

Reply:
I copied the file from the adaptor library. I assumed it was right.

Reply:
The full code was in the first commit of the pull request, but we backed out the changes the library itself didn't need. Not sure the easiest way to grab these, either checkout the commit or just copy&paste since it's just a few blocks of code, including the tests.

https://github.com/googlegsa/library/pull/83/commits/6c3dee8e278e9f4a89f7a34b4361bbbcacf0e355